### PR TITLE
Allow no-interaction option to generate the files

### LINKED ISSUE #352: Write files even if no-interaction option is set
It'd be nice to generate all the defaults if --no-interaction is sent to the bin/console make:user:msgphp command

    bin/console make:user:msgphp --no-interaction

### PR DESCRIPTION
**Pull request by tacman**
I _think_ this should work, you still need to install the Differ bundle, even though it's not actually essential.  Ideally those file diffs wouldn't be displayed, but $input isn't available in the method that checks for that bundle.

This option has no tests right now.

fixes #352

### CODE REVIEW

**Comment by ro0NL** [comment]
i think defaulting `$review` to false in no-interaction mode is the easiest fix :)

https://github.com/msgphp/msgphp/blob/fddcbf809d32acb3971b5bd7355eecb799af43dc/src/UserBundle/Maker/UserMaker.php#L123

File: src/UserBundle/Maker/UserMaker.php [hunk 1]
```diff
@@ -157,12 +157,12 @@ public function generate(InputInterface $input, ConsoleStyle $io, Generator $gen
                 } else {
                     $io->writeln($differ->diff($exist ? file_get_contents($file) : '', $contents));
                 }
-            } while ('r' === $choice = $io->choice('Write changes and continue reviewing?', $choices, $choices['r']));
+            } while ( !$input->getOption('no-interaction') || ('r' === $choice = $io->choice('Write changes and continue reviewing?', $choices, $choices['r'])) );
```
Comment (ro0NL) [inline src/UserBundle/Maker/UserMaker.php]: i suggest to rewrite `$review` above like

`$review = $io->confirm('...', $input->isInteractive());`

**Comment by tacman** [comment]
I'm thinking now instead the default should be choices['y'], then it'll be
consistent with simply hitting return multiple times.

Also, the "review files" question I think needs to default to "no" if it's
interactive, otherwise it dumps everything on the screen.  But again, I'd
prefer to have that default be no anyway.

**Comment by ro0NL** [comment]
personally i prefer in interactive a mode to let users opt-out explicitly from reviewing, we should stay on the safe side.

in no-interaction mode Symfony will automatically pick the default value for things like `confirm()` etc. So my previous suggesting ensures `$review=false` in no-interaction mode, as such we'll never hit the file reviewing part.

**Comment by tacman** [comment]
After I've selected the important things (allow registration, etc.), then I
pretty much start hitting return as fast as I can to get to the end.  So my
preference is that it wouldn't stop and have me review and confirm changes,
especially if it's the first time the file is being written.

But I think the no-interaction will help a lot for simply creating a
working demo, not just of this app, but a working demo of other apps and
bundles.

**Comment by ro0NL** [comment]
additionally we could add a `--no-review` option :)

**Review by ro0NL — APPROVED** [review]
@tacman ive pushed some commits to finish this one. All good now, thanks for pointing this out.

**Pull request merged**